server/game/gameSystems/DelayedEffectSystem.ts

```diff
@@ -37,36 +43,38 @@ export class DelayedEffectSystem<TContext extends AbilityContext = AbilityContex
         //     properties.ability = event.context.ability;
         // }
 
-        const { when, duration, limit, immediateEffect, ...otherProperties } = properties;
+        const { title, when, duration, limit, immediateEffect, ...otherProperties } = properties;
 
-        const delayedEffect = AbilityHelper.ongoingEffects.delayedEffect({
-            title: 'Discard a card from your hand',
-            when: when,
-            immediateEffect: immediateEffect,
-            limit: limit
-        });
+        const renamedProperties = { ...otherProperties, ongoingEffect:
+            AbilityHelper.ongoingEffects.delayedEffect({
+                title,
+                when,
+                immediateEffect,
+                limit
+            }) };
 
-        const renamedProperties = Object.assign(otherProperties, { ongoingEffect: delayedEffect });
+        const delayedEffectTarget = properties.effectType === DelayedEffectType.Card ? event.context.target : event.context.source;
```

Comment:
I realized that calling it `delayedEffectTarget` is a little misleading since it's really the card that is registering the ability, not the target of any effects. Maybe `delayedEffectSource`?

Also, this made me wonder - have we done any tests where the `delayedEffectTarget` unit gets defeated? I'm trying to understand why it's even relevant which card is registering the effect, I'm _pretty_ sure it's not because the effect goes away if the unit leaves the field but it's worth confirming that.

Reply:
@AMMayberry1 I already had a test for this in Change of Heat, I think. LMK if you think an additional test is needed or what scenario we could use to further test this

Reply:
Would be great to also do one with Zorii if we don't have it already, where she gets defeated before the end of phase effect happens

Reply:
Ah yeah that has been there since the beginning